### PR TITLE
feat(vue): 🎸 added markers, autoResizeCanvas and playOnHover props

### PR DESCRIPTION
**Pull request by afsalz**
## Description
- feat(vue): 🎸 added markers, animationId, themeId, themeData, autoResizeCanvas and playOnHover props
- perf(vue): ⚡️ added setViewport to render visible canvas

## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] This is something we need to do.

### CODE REVIEW

File: packages/vue/src/dotlottie.ts [hunk 1]
```diff
@@ -128,22 +173,37 @@ export const DotLottieVue = defineComponent({
...
 
       intersectionObserver = getIntersectionObserver();
-      resizeObserver = getResizeObserver();
       intersectionObserver.observe(canvas.value);
-      resizeObserver.observe(canvas.value);
+      if (typeof autoResizeCanvas?.value === 'boolean' && autoResizeCanvas.value) {
+        resizeObserver = getResizeObserver();
+        resizeObserver.observe(canvas.value);
+      }
       document.addEventListener('visibilitychange', onVisibilityChange);
```
Comment (theashraf) [inline packages/vue/src/dotlottie.ts:194]: we don't need to freeze on visibility change, as the browser automatically stops the animation loop when the tab is not active. Maybe review the React implementation to match Vue

Reference: https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API#policies_in_place_to_aid_background_page_performance

File: packages/vue/src/dotlottie.ts [hunk 2]
```diff
@@ -32,11 +35,24 @@ export const DotLottieVue = defineComponent({
     speed: { type: Number, required: false },
     src: { type: String, required: false },
     useFrameInterpolation: { type: Boolean, required: false },
+    marker: { type: String, required: false },
+    autoResizeCanvas: { type: Boolean, required: false, default: true },
+    playOnHover: { type: Boolean, required: false },
```
Comment (theashraf) [inline packages/vue/src/dotlottie.ts:40]: how about `themeId` & `animationId` ? any other props left not implemented ?

**Review by theashraf — COMMENTED** [review]
There are two rendering optimizations in react that we can also bring to vue. 

1. partial rendering using setViewport API in dotlottie-web
2. initially freeze of the animation if it's initially outside of viewport

File: .changeset/fast-ghosts-teach.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@lottiefiles/dotlottie-vue': minor
+---
+
+feat: 🎸 added markers, autoResizeCanvas and playOnHover
```
Comment (theashraf) [inline .changeset/fast-ghosts-teach.md:5]:
```suggestion
feat(vue): 🎸 added markers, autoResizeCanvas and playOnHover props
```

File: packages/vue/src/dotlottie.ts (hunk 1 above)
Comment (afsalz) [reply packages/vue/src/dotlottie.ts]: removed

File: packages/vue/src/dotlottie.ts (hunk 2 above)
Comment (afsalz) [reply packages/vue/src/dotlottie.ts:44]: Sorry, was following an old commit of react player. Added the latest updates. animationId, themeId and themeData

**Review by theashraf — APPROVED** [review]
🚀

**Review by theashraf — APPROVED** [review]

**Pull request merged**